.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,64 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
```

Comment:
I wonder if we should run it also nightly on main. You know, just in case one does a `git push` in the wrong direction :P